### PR TITLE
Fix attempt for activity mail gmail threading fail

### LINKED ISSUE #5015: Developer replies are not threaded
### Describe the problem and steps to reproduce it:

1) Receive two developer replies or other messages on one add-on

### What happened?
* The emails show up as two separate threads in my email client

### What did you expect to happen?
* Replies should include a References header to make them show up as one thread in my email client.

### PR DESCRIPTION
**Pull request by eviljeff**
Another attempt to fix #5015 by enclosing references and in-reply-to headers in `< >`

### CODE REVIEW

File: src/olympia/activity/tests/test_utils.py [hunk 1]
```diff
@@ -516,7 +516,7 @@ def test_send_activity_mail():
     assert len(mail.outbox) == 1
     assert mail.outbox[0].body == message
     assert mail.outbox[0].subject == subject
-    reference_header = '{addon}/{version}@{site}'.format(
+    reference_header = '<{addon}/{version}@{site}>'.format(
```
Comment (muffinresearch) [inline src/olympia/activity/tests/test_utils.py]: Does `<` or `>` need to be escaped in the content?

Comment (eviljeff) [inline src/olympia/activity/tests/test_utils.py]: not as far as I'm aware

Comment (EnTeQuAk) [inline src/olympia/activity/tests/test_utils.py]: This is similar to `Christopher Grebs <cg@webshox.org>` wrapping, doesn't and should be escaped.

Comment (eviljeff) [inline src/olympia/activity/tests/test_utils.py]: I don't know how to parse that comment @EnTeQuAk 😖

Comment (wagnerand) [inline src/olympia/activity/tests/test_utils.py]: @EnTeQuAk This is not a `From` or `To` header. It's used only in `References` and `In-Reply-To` which take message-ids only.

**Comment by eviljeff** [comment]
I still don't know what @EnTeQuAk meant, but I got two r+ so merging.

File: src/olympia/activity/tests/test_utils.py (hunk 1 above)
Comment (EnTeQuAk) [inline src/olympia/activity/tests/test_utils.py]: looking at many other `References` and `In-Reply-To` headers this is perfectly valid, there's other resources like https://cr.yp.to/immhf/thread.html that mention it. I fail to find the relevant place in the RFC though, email RFCs are bad.

**Pull request merged**